src/server/controllers/users.controller.js

```diff
@@ -212,102 +211,83 @@ export function getJobCategory(req, res, next) {
-        if (err) {
-          reject(false);
+    User.findOne({ _id: req.session._id }).exec()
+      .then(user => {
+        if (user) {
+          return validateEmail(req.body.email);
         } else {
-          file = files.image; // file when postman test.
-          field = JSON.parse(unescape(fields.info));
-          editData = {
-            name: field.name,
-            languages: field.languages,
-            location: field.location,
-            about: field.about,
-            education: field.education,
-            work: field.work,
-          };
-          resolve(true);
+          throw err;
         }
-      });
-    })
-    .then(parsed => {
-      if (parsed) {
-        return User.findOne({ _id: req.session._id, email: { $ne: null } }).exec();
-      } else {
-        throw new err;
-      }
-    })
-    .then(userWithEmail => {
-      if (!userWithEmail) {
-        if (field.email === null) {
-          res.status(400).json({ err_point: userCallback.ERR_INVALID_UPDATE });
+      })
+      .then((isValid) => {
+        if (isValid) {
+          let editData = {
+            name: req.body.name,
+            email: req.body.email,
+            languages: req.body.languages,
+            location: req.body.location,
+            about: req.body.about,
+            education: req.body.education,
+            work: req.body.work,
+          };
+          return User.update({ _id: req.session._id }, { $set: editData }).exec();
         } else {
-          validateEmail(field.email)
-            .then((isValid) => {
-              return User.update({ _id: req.session._id },
-                { $set: { email: field.email }, editData }).exec();
-            });
+          throw new Error(userCallback.ERR_INVALID_MAIL);
         }
-      } else {
-        return User.update({ _id: req.session._id }, { $set: editData }).exec();
-      }
-    })
-    .then(updateData => {
-      if (updateData) {
-        return setKey();
-      } else {
-        throw new err;
-      }
-    })
-    .then(data => {
-      if (data) {
-        if (file === undefined) {
-          res.status(200).json({ msg: userCallback.SUCCESS_UPDATE_WITHOUT_IMAGE });
+      })
+      .then(updateData => {
+        if (updateData) {
+          return setKey();
         } else {
-          let bucketName = 'yodabucket';
-          let imageKey = `profile/${req.session._id}.png`;
-          let readStream = fs.createReadStream(file.path);
+          throw new Error(userCallback.ERR_MONGOOSE);
+        }
+      })
+      .then(keyData => {
+        if (keyData) {
+          if (req.body.image == null) {
+            console.log(req.body.image);
+            res.status(200).json({ msg: userCallback.SUCCESS_UPDATE_WITHOUT_IMAGE });
+          } else {
+            console.log(req.body.image);
+            let bucketName = 'yodabucket';
+            let now = new Date();
+            let imageKey = `profile/${req.session._id}/${now.getTime()}/.png`;
```

Comment:
We make each user profile folder with their id and each folder contains images named with timestamp. Maybe using '_' would be good also, but previous one would be better to manage lots of file. Thanks!